lib/receivers/tcpReceiver.js

```diff
@@ -13,6 +13,21 @@
 const net = require('net');
 const log = require('../log/logger');
 
+function getSerializableError(err) {
+  let errorToLog = err.toString();
+
+  if (!(err instanceof Error)) {
+    try {
+      JSON.stringify(err); // if err has circular references, this would fail
+      errorToLog = err;
+    } catch (err) {
+      errorToLog = 'Error argument not instance of Error and not stringifiable';
```

Comment:
Should we add some local logging around these special cases?

Reply:
Not sure I follow you @samlevin

Reply:
I think he means something like `log.debug('Task parsing complete');`

Reply:
Done (if I understood correctly).